zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/ScaleUpPartitionsTest.java

```diff
@@ -321,6 +328,81 @@ public void shouldBePossibleToRestoreAsAfterScalingUp(final boolean backupBefore
     }
   }
 
+  @Test
+  public void shouldNotBreakMessageCorrelation() {
+    // given
+    cluster.awaitHealthyTopology();
+    final var messageId = new AtomicInteger();
+    final var correlationKeyVariable = "correlationKey";
+    final var correlationKeys = new ArrayBlockingQueue<Integer>(1024);
+    final var processId = "PROCESS_WITH_MESSAGE";
+    final var process =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .intermediateCatchEvent(
+                "catch-me",
+                b ->
+                    b.message(
+                        m ->
+                            m.name("message")
+                                .zeebeCorrelationKeyExpression(correlationKeyVariable)))
+            .endEvent()
+            .done();
+
+    final var deploymentKey =
+        camundaClient
+            .newDeployResourceCommand()
+            .addProcessModel(process, "process.bpmn")
+            .send()
+            .join()
+            .getKey();
+    new ZeebeResourcesHelper(camundaClient).waitUntilDeploymentIsDone(deploymentKey);
+    final var targetPartitionCount = PARTITIONS_COUNT + 1;
+
+    final Supplier<Map<String, Object>> variableProvider =
+        () -> {
+          final var key = messageId.incrementAndGet();
+          if (!correlationKeys.add(key)) {
+            throw new IllegalStateException("Too many correlation keys created");
+          }
+          return Map.of(correlationKeyVariable, key);
+        };
+
+    createInstanceWithAJobOnAllPartitions(
+        camundaClient, JOB_TYPE, PARTITIONS_COUNT, false, processId, variableProvider);
```

Comment:
:x: The naming got me confused. We don't want to create instance with jobs, instead create instance with message catch event. The instances of the above deployed process must be created. May be you can refactor the util method?

Reply:
I thought splitting the method into two allow "only" creating instances. This way it is more reusable. Jobtype is only needed if it has to deploy a process.

```
createInstancesOnAllPartitions(partitionCount, processID, variables) { ... }

createInstanceWithJobsOnAllpartitions(...){
    deployprocess();
    createInstancesOnAllPartitions(...)
```